src/style-spec/expression/index.js

```diff
@@ -135,37 +107,65 @@ function createExpression(expression: mixed,
         };
     }
 
+    return {
+        result: 'success',
+        evaluate,
+        parsed
+    };
+}
+
+type ConstantExpression = {
+    result: 'constant',
+    evaluate: (globals: GlobalProperties, feature?: Feature) => any
+};
+
+type SourceExpression = {
+    result: 'source',
+    evaluate: (globals: GlobalProperties, feature?: Feature) => any,
+};
```

Comment:
What do you think about `FeatureExpression`?  Seems more direct to me, but maybe there was a reason we opted for `SourceFunction` over `FeatureFunction` before...

Reply:
#5531 suggests `DataExpression` -- let's hash out terminology there. For now, I'm happy just getting JS consistent with current native terms. 😄